wingtips-zipkin/src/main/java/com/nike/wingtips/zipkin/util/ZipkinSpanSenderDefaultHttpImpl.java

```diff
@@ -0,0 +1,203 @@
+     * <p>
+     *     This code was derived from the Zipkin/Brave repository's brave-spancollector-http module v3.9.1
+     *     (https://github.com/openzipkin/brave/blob/master/brave-spancollector-http/src/main/java/com/github/kristofa/brave/http/HttpSpanCollector.java)
+     *     and licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0).
+     * </p>
+     */
+    protected void sendSpans(byte[] spanListJsonPayloadBytes) throws IOException {
+        logger.trace("Sending spans to zipkin");
+
+        // intentionally not closing the connection, so as to use keep-alives
+        HttpURLConnection connection = (HttpURLConnection) postZipkinSpansUrl.openConnection();
+        connection.setConnectTimeout(connectTimeoutMillis);
+        connection.setReadTimeout(readTimeoutMillis);
+        connection.setRequestMethod("POST");
+        connection.addRequestProperty("Content-Type", "application/json");
+        if (compressZipkinSpanPayload) {
+            connection.addRequestProperty("Content-Encoding", "gzip");
+            ByteArrayOutputStream gzipped = new ByteArrayOutputStream();
+            try (GZIPOutputStream compressor = new GZIPOutputStream(gzipped)) {
+                compressor.write(spanListJsonPayloadBytes);
+            }
+            spanListJsonPayloadBytes = gzipped.toByteArray();
+        }
+        connection.setDoOutput(true);
+        connection.setFixedLengthStreamingMode(spanListJsonPayloadBytes.length);
+        connection.getOutputStream().write(spanListJsonPayloadBytes);
+
+        try (InputStream in = connection.getInputStream()) {
+            while (in.read() != -1) ; // skip
+        } catch (IOException e) {
+            try (InputStream err = connection.getErrorStream()) {
+                if (err != null) { // possible, if the connection was dropped
+                    while (err.read() != -1) ; // skip
+                }
+            }
+            throw e;
+        }
+    }
+
+    protected static class ZipkinSpanSenderJob implements Runnable {
+        private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+        protected final ZipkinSpanSenderDefaultHttpImpl zipkinSpanSender;
+        protected final BlockingQueue<zipkin.Span> zipkinSpanSendingQueue;
+
+        public ZipkinSpanSenderJob(ZipkinSpanSenderDefaultHttpImpl zipkinSpanSender, BlockingQueue<zipkin.Span> zipkinSpanSendingQueue) {
+            this.zipkinSpanSender = zipkinSpanSender;
+            this.zipkinSpanSendingQueue = zipkinSpanSendingQueue;
+        }
+
+        @Override
+        public void run() {
+            try {
+                if (zipkinSpanSendingQueue.isEmpty())
+                    return;
+
+                List<zipkin.Span> drainedSpans = new ArrayList<>(zipkinSpanSendingQueue.size());
+                zipkinSpanSendingQueue.drainTo(drainedSpans);
+                if (!drainedSpans.isEmpty())
+                    zipkinSpanSender.sendSpans(drainedSpans);
```

Comment:
I took a quick look at what you're proposing for zipkin-reporter-java and I definitely like the idea. I'll probably follow up this release with a minor refactor to use zipkin-reporter-java's impl once that's ready to go.